tools/rpkg/src/types.cpp

```diff
@@ -40,7 +40,7 @@ Vector RType::GetFactorLevels() const {
 	D_ASSERT(id_ == RTypeId::FACTOR);
 	Vector duckdb_levels(LogicalType::VARCHAR, aux_.size());
 	auto levels_ptr = FlatVector::GetData<string_t>(duckdb_levels);
-	for (R_xlen_t level_idx = 0; level_idx < aux_.size(); level_idx++) {
+	for (R_xlen_t level_idx = 0; (size_t)level_idx < aux_.size(); level_idx++) {
```

Comment:
No reason to use `R_xlen_t` at all:

```suggestion
	for (size_t level_idx = 0; level_idx < aux_.size(); ++level_idx) {
```